### ISSUE TITLE
TensorFlow 1.3.0 GPU for Windows doesn't work for 2 NVIDIA P4s

### ISSUE THREAD

**Issue by johnsrude** [tensorflow/tensorflow#13692]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 7 (factory production machine)
- **TensorFlow installed from (source or binary)**:
https://pypi.python.org/pypi/tensorflow-gpu
tensorflow_gpu-1.3.0-cp35-cp35m-win_amd64.whl

- **TensorFlow version (use command below)**: 1.3.0
- **Python version**:  3.5
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: Cuda release 8.0, V8.0.60.  cuDNN 6.
- **GPU model and memory**:  2 NVIDIA Tesla P4s (same as specs)
- **Exact command to reproduce**: https://github.com/tkuanlun350/Tensorflow-SegNet

### Describe the problem
We're currently evaluating Tensorflow as a possible replacement for Caffe but we seem to have run into a problem with running multiple P4s on a production system.
 
Our production systems are offline so I had to install a Python wheel of the GPU version TF framework which works fine until I try to use 2 GPUs for training where I get the error message:
 
`2017-10-05 13:53:42.989423: I C:\tf_jenkins\home\workspace\rel-win\M\windows-gpu\PY\35\tensorflow\core\common_runtime\gpu\gpu_device.cc:847] Peer access not supported between device ordinals 1 and 0`

P4s are installed and working correctly.  I don't understand why peer-to-peer GPU access is required for using GPUs.  We can run multiple GPUs on our Caffe implementation.

```
nvidia-smi
Tue Oct 10 10:53:57 2017
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 385.54                 Driver Version: 385.54                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P4            TCC  | 00000000:04:00.0 Off |                    0 |
| N/A   31C    P8     6W /  75W |      0MiB /  7590MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P4            TCC  | 00000000:81:00.0 Off |                    0 |
| N/A   30C    P8     6W /  75W |      0MiB /  7590MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

```

**Comment by mrry** [comment]
The log message in your report is not an error message, but rather an [information message](https://github.com/tensorflow/tensorflow/blob/624bcfe409601910951789325f0b97f520c0b1ee/tensorflow/core/common_runtime/gpu/gpu_device.cc#L847). (I believe it's been removed in TF 1.4, because it was confusing people more than it enlightened them.)

You should be able to ignore the message and continue to use TensorFlow on your GPUs. Did the computation fail at a later point?

**Comment by johnsrude** [comment]
I'll let you know after this issue is reconciled:
https://stackoverflow.com/questions/46697662/repairing-tensorflow-checkpoint

**Comment by mrry** [comment]
Is it possible to run one of the example programs (e.g. from https://github.com/tensorflow/models/) in the meantime?

**Comment by johnsrude** [comment]
Okay, looks like both of them work for training.  Thanks for your help.

The Nettles: Celtic. Eclectic. Folkadelic.
www.TheNettles.com <http://www.thenettles.com/>

**Comment by johnsrude** [comment]
@mrry @aselle  Please reopen this issue. 

It turns out that this bug is still in place because while computation doesn't fail the number of images processed per second is exactly the same whether there are one or two P4s enabled in the machine which would seem to indicate that only card is being used.

**Comment by mrry** [comment]
Does the code you are running place ops on the different GPUs? (If it doesn't include explicit code for parallelizing across the GPUs, then I would expect it to use `/gpu:0` only.)

**Comment by johnsrude** [comment]
The answer to  your question is "no".  

I've read https://www.tensorflow.org/tutorials/deep_cnn#training_a_model_using_multiple_gpu_cards

Should I post my questions about how to do multi-GPU training on StackOverflow?

**Comment by johnsrude** [comment]
I notice that this issue is still open:
https://github.com/google/seq2seq/issues/44

**Comment by bshao001** [comment]
If you want to use model parallelism, refer to the TensorFlow NMT model tutorial. However, based on my limited test, separating a seq2seq model (NMT model) in two layers to 2 GPUs does not speed up the training at all. It actually makes the training slightly slower. Running a 4-layer model might help using that. However, I could not see any capacity increase by increasing the number of layers (you can clearly see the capacity increase when you increase the number of units in each layer). Therefore, data parallelism is still requested by many users like me. I have one issue open here: https://github.com/tensorflow/nmt/issues/141, but was never replied there.

Another interesting fact I would like to mention here is: If you only have 2 GPUs in one machine, you do not need to try any so-called parallelism, because setting colocate_gradients_with_ops in function tf.gradients to false cuts down the training time to half, while this requires to be true for any parallelism using TensorFlow. I am not sure why that makes such a big difference.

**Comment by johnsrude** [comment]
Tried setting colocate_gradients_with_ops to True but it didn't make a difference in performance in my case. YMMV.

**Comment by bshao001** [comment]
@johnsrude The nmt model original implementation was in True. I set it to False, and cut down the training time to half.

**Comment by johnsrude** [comment]
Oh.  Well it's set to False by default so I guess it doesn't work for this model.
